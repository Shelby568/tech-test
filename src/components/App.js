import React, { useState } from "react";
import "../styles/App.css";
import Search from './Search';
import SearchResults from './SearchResults';

function App() {
  const [searchResults, setSearchResults] = useState([]);
  const [searchText, setSearchText] = useState('');
  
  return (
    <div className="App">
      <img className="logo" src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg" alt="NASA Logo" />
      <Search setSearchResults={ setSearchResults } searchText={searchText} setSearchText={setSearchText}/>
      {searchResults.length > 0 && <SearchResults results={searchResults}/>}
      {searchText.length > 1 && !searchResults.length && <p className="no-results">No results</p>}
   
    </div>
  );
}

export default App;
