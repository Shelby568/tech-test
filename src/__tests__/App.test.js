import React from 'react';
import { render } from '@testing-library/react';
import App from '../components/App';

describe('App', () => {
    const { asFragment } = render(<App />)
    
    it('makes sure component renders correctly', () => {
        expect(asFragment()).toMatchSnapshot();
    });
});