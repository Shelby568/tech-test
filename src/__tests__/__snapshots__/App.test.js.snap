// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App makes sure component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="NASA Logo"
      class="logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <form
      class="searchform"
    >
      <input
        class="searchbox"
        placeholder="Search for images.."
        type="text"
      />
      <button
        class="submit"
        type="submit"
      >
        Search
      </button>
    </form>
    <img
      alt="test"
      class="test-image"
      src="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
    />
  </div>
</DocumentFragment>
`;
